<launch>
    <include file="$(find liv_slam)/launch/vlp_16_slam.launch"/>
    
    <!-- run pointcloud_to_laserscan node -->
    <node pkg="liv_slam" type="Pointcloud2laserscan_node" name="Pointcloud2laserscan_node">
        <!--
        <remap from="cloud_in" to="/rslidar_points"/>
        <remap from="scan" to="/rslidar/scan"/>
        -->
        <rosparam>
            # target_frame: rslidar # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: -0.4
            max_height: 1.0
 
            angle_min: -3.1415926 # -M_PI
            angle_max: 3.1415926 # M_PI
            angle_increment: 0.003 # 0.17degree
            scan_time: 0.1
            range_min: 0.2
            range_max: 50
            use_inf: true
            inf_epsilon: 1.0
 
            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>
    </node>

    <node pkg="liv_slam" type="2d_GridMapping" name="GridMapping_node" output="screen" clear_params="true">
		<rosparam file="$(find liv_slam)/config/default.yaml" command="load" />
    </node>

</launch>